src/mgmt/srv_pool.c

```diff
@@ -668,8 +716,42 @@ ds_mgmt_pool_target_update_state(uuid_t pool_uuid, uint32_t rank,
 {
 	int			rc;
 	d_rank_list_t		*ranks;
+	d_rank_list_t		*reint_ranks = NULL;
 	struct mgmt_svc		*svc;
 
+	if (state == PO_COMP_ST_UP) {
+		/* When doing reintegration, need to make sure the pool is
+		 * created and started on the target rank
+		 */
+
+		reint_ranks = d_rank_list_alloc(1);
+		if (reint_ranks == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+
+		reint_ranks->rl_nr = 1;
+		reint_ranks->rl_ranks[0] = rank;
```

Comment:
It looks like you can just use local variable here. like
d_rank_list_t reint_rank;
d_rank_t rank
reint_rank.rl_nr = 1;
reint_ranks.ranks = &rank; ?

And also reint_ranks can be inside this {} ?

Reply:
Good idea, fixed